### PR TITLE
312: remove unnecessary iterations on successful payment processing

### PR DESCRIPTION
**Pull request by JudeNiroshan**
## Description

- Fixed unnecessary iterations on successful payment handling
- Reduce retry count from 20 to 5
- Add variable delay for retrying mechanism

### CODE REVIEW

File: service/src/main/java/com/commercetools/pspadapter/payone/PaymentHandler.java [hunk 1]
```diff
@@ -73,7 +73,8 @@ public PaymentHandleResult handlePayment(@Nonnull final String paymentId) {
         } catch (final Exception e) {
             return handleException(e, paymentId);
```
Comment (ahalberkamp) [inline service/src/main/java/com/commercetools/pspadapter/payone/PaymentHandler.java:74]: Does it make sense to log the retrycounter here too

Comment (JudeNiroshan) [reply service/src/main/java/com/commercetools/pspadapter/payone/PaymentHandler.java]: @ahalberkamp yeah, totally. My previous git push has missed some changelogs. Please check now :)

**Review by ahalberkamp — APPROVED** [review]

File: service/src/main/java/com/commercetools/pspadapter/payone/PaymentHandler.java [hunk 2]
```diff
@@ -51,29 +52,30 @@ public PaymentHandler(String payoneInterfaceName, String tenantName,
      * @return the result of handling the payment
      */
     public PaymentHandleResult handlePayment(@Nonnull final String paymentId) {
-        PaymentHandleResult result = null;
+        int retryCounter = 0;
         try {
-            for (int i = 0; i < RETRIES_LIMIT; i++) {
+            for (; retryCounter < RETRIES_LIMIT; retryCounter++) {
```
Comment (ninalindl) [inline service/src/main/java/com/commercetools/pspadapter/payone/PaymentHandler.java:57]: Isn't the first semicolon unnecessary?

**Review by ninalindl — APPROVED** [review]

File: service/src/main/java/com/commercetools/pspadapter/payone/PaymentHandler.java (hunk 2 above)
Comment (JudeNiroshan) [reply service/src/main/java/com/commercetools/pspadapter/payone/PaymentHandler.java:57]: it's good you point it out. Unfortunately we need this. In java for loop, we must have a statement/assignment as the first parameter to the for loop. Therefore we cannot remove that semi-colon. We can do it in GoLang 😄

**Pull request merged**